### PR TITLE
Fix notification tools

### PR DESCRIPTION
**Pull request by danieljurek**
Set TargetFramework netcoreapp2.0 -> netcoreapp3.1 so it builds on devops machines. There is no specific reason to keep using older versions.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

**Comment by danieljurek** [comment]
Successful pipeline run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=701252&view=results

**Comment by danieljurek** [comment]
@weshaggard -- Couple small changes to ensure we fetch all groups and ensure group names are an appropriate length

**Comment by mitchdenny** [comment]
You could go .NET 5.0 😱

**Comment by danieljurek** [comment]
Suggestions from Wes: 

* If a pipeline name is 64 characters or more, throw an error 
* File an issue to make the naming scheme more resilient

File: tools/notification-configuration/notification-creator/NotificationConfigurator.cs [hunk 1]
```diff
@@ -83,7 +94,9 @@ public NotificationConfigurator(AzureDevOpsService service, ILogger<Notification
                 var newTeam = new WebApiTeam
                 {
                     Description = YamlHelper.Serialize(teamMetadata),
-                    Name = $"{pipeline.Name} -- {suffix}",
+                    // Ensure team name fits within maximum 64 character limit
+                    // https://docs.microsoft.com/en-us/azure/devops/organizations/settings/naming-restrictions?view=azure-devops#teams
+                    Name = StringHelper.MaxLength($"{pipeline.Name} -- {suffix}", 64),
```
Comment (weshaggard) [inline tools/notification-configuration/notification-creator/NotificationConfigurator.cs:99]:
```suggestion
                    Name = StringHelper.MaxLength($"{pipeline.Name} -- {suffix}", MaxTeamNameLength),
```

**Review by weshaggard — APPROVED** [review]
One suggestion but looks good.

**Review by mitchdenny — COMMENTED** [review]
LGTM

**Pull request merged**